CromIAM/src/main/scala/cromiam/instrumentation/CromIamInstrumentation.scala

```diff
@@ -0,0 +1,65 @@
+package cromiam.instrumentation
+
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import cats.data.NonEmptyList
+import cromwell.core.instrumentation.InstrumentationPrefixes._
+import cromwell.services.instrumentation.CromwellInstrumentation
+import cromwell.services.instrumentation.CromwellInstrumentation.InstrumentationPath
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object CromIamInstrumentation {
+  private val UUIDRegex = {
+    val hex = """[a-fA-F\d]"""
+    s"""$hex{8}-$hex{4}-$hex{4}-$hex{4}-$hex{12}"""
+  }
+}
+
+trait CromIamInstrumentation extends CromwellInstrumentation {
+
+  val samPrefix: NonEmptyList[String] = NonEmptyList.one("sam")
+  val getWhitelistPrefix = NonEmptyList.one("get-whitelist")
+  val userCollectionPrefix = NonEmptyList.one("user-collection")
+  val authCollectionPrefix = NonEmptyList.one("auth-collection")
+  val registerCollectionPrefix = NonEmptyList.one("register-collection")
+  val rootWfIdPrefix = NonEmptyList.one("root-workflow-id")
+  val wfCollectionPrefix = NonEmptyList.one("workflow-collection")
+
+
+  def convertRequestToPath(httpRequest: HttpRequest): NonEmptyList[String] = NonEmptyList.of(
+    // Returns the path of the URI only, without query parameters (e.g: api/engine/workflows/metadata)
+    httpRequest.uri.path.toString().stripPrefix("/")
+      // Replace UUIDs with [id] to keep paths unique regardless of the workflow
```

Comment:
isn't this to keep paths the _same_ regardless of the workflow ID?

Reply:
Yes! Changed it.